dev/com.ibm.ws.microprofile.config.1.4/src/com/ibm/ws/microprofile/config14/impl/ConfigAccessorImpl.java

```diff
@@ -0,0 +1,222 @@
+    private Converter<T> converter;
+    private T defaultValue;
+    private String defaultString;
+    private long cacheFor;
+    private TimeUnit cacheForUnit;
+    private String suffixValue;
+    private ConfigAccessor<String> suffixAccessor;
+    private boolean evaluateVariables;
+    private WebSphereConfig config;
+
+    public ConfigAccessorImpl(WebSphereConfig config, String propertyName, Class<T> conversionType) {
+        this(config, propertyName, conversionType, true); //evaluateVariables defaults to true
+    }
+
+    public ConfigAccessorImpl(WebSphereConfig config, String propertyName, Class<T> conversionType, boolean evaluateVariables) {
+        this.config = config;
+        this.propertyName = propertyName;
+        this.conversionType = conversionType;
+        this.evaluateVariables = evaluateVariables;
+    }
+
+//    /** {@inheritDoc} */
+//    @Override
+//    public <N> ConfigAccessor<N> as(Class<N> clazz) {
+//        ConfigAccessor<N> accessor = new ConfigAccessorImpl<>(propertyName, clazz);
+//        return accessor;
+//    }
+//
+//    /** {@inheritDoc} */
+//    @Override
+//    public ConfigAccessor<List<T>> asList() {
+//        ConfigAccessor<List<T>> accessor = new ConfigListAccessorImpl<>(propertyName, conversionType);
+//        return accessor;
+//    }
+//
+//    /** {@inheritDoc} */
+//    @Override
+//    public ConfigAccessor<Set<T>> asSet() {
+//        ConfigAccessor<Set<T>> accessor = new ConfigSetAccessorImpl<>(propertyName, conversionType);
+//        return accessor;
+//    }
+//
+//    /** {@inheritDoc} */
+//    @Override
+//    public ConfigAccessor<T> useConverter(Converter<T> converter) {
+//        this.converter = converter;
+//        return this;
+//    }
+//
+//    /** {@inheritDoc} */
+//    @Override
+//    public ConfigAccessor<T> withDefault(T value) {
+//        this.defaultValue = value;
+//        return this;
+//    }
+//
+//    /** {@inheritDoc} */
+//    @Override
+//    public ConfigAccessor<T> withStringDefault(String value) {
+//        this.defaultString = value;
+//        return this;
+//    }
+
+    /** {@inheritDoc} */
+    @Override
+    public ConfigAccessor<T> cacheFor(long value, TimeUnit timeUnit) {
+        this.cacheFor = value;
+        this.cacheForUnit = timeUnit;
+        return this;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public ConfigAccessor<T> evaluateVariables(boolean evaluateVariables) {
+        this.evaluateVariables = evaluateVariables;
+        return this;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public ConfigAccessor<T> addLookupSuffix(String suffixValue) {
+        this.suffixValue = suffixValue;
+        return this;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public ConfigAccessor<T> addLookupSuffix(ConfigAccessor<String> suffixAccessor) {
+        this.suffixAccessor = suffixAccessor;
+        return this;
+    }
+
+    public static String resolve(Config config, String raw) {
```

Comment:
A couple of checks on this method

I guess the spec isn't finalised here but just to check
* The string should be repeatedly processed until there are no more variable references?
* There's no escaping for if the user wants to include a literal `${`?

Can I do something dumb like this:
```
foo = "${bar}wibble${baz}"
bar = "${"
baz = "}"
wibble = "qux"
```
and have `foo` resolve to `"qux"?

What's the expected behaviour if I reference a variable that doesn't exist?

Reply:
is this a spec question? i don't think there is an escape character defined yet ... best you can do right now is to turn off resolution completely.

Reply:
How about this:

```
foo = "${bar}wibble${baz}"
bar = "${"
baz = "${fuz}"
fuz = "yay"
wibble = "qux"
```

This will try to resolve `wibble${fuz` and fail.

Reply:
Your other approach would be to iterate through the string, finding the variables, recursively resolve each of them, and then stick the string back together.